### DISCUSSION TITLE
How to determine whether a raft node is the leader?

### DISCUSSION THREAD

**Discussion by garan82** [etcd-io/etcd#14265] [Q&A] [answered]
In my use case, there is some work that should only be done once and the result would be propagated to other nodes. How can I make sure that only one node does this work? I am thinking of having the leader to do it, but I dont see any doc on determining whether a raft node is the leader. Any suggestions? Thanks!

**Comment by mind1949** [comment, marked as answer]
> In my use case, there is some work that should only be done once and the result would be propagated to other nodes. How can I make sure that only one node does this work? I am thinking of having the leader to do it, but I dont see any doc on determining whether a raft node is the leader. Any suggestions? Thanks!

If you are using raft package, `Node.Status().RaftState.RaftState == StateLeader` is your choice .

**Comment by SimFG** [comment]
There is an example in the benchmark. Hope it helps you.
https://github.com/etcd-io/etcd/blob/4977877730cf341d8ef2daf321c87b411a2022d1/tools/benchmark/cmd/util.go#L43-L67

**Reply by garan82** [reply]
Thank you all!
